### ISSUE TITLE
New install, requiring a supplier to create asset

### ISSUE THREAD

**Issue by alockbox** [grokability/snipe-it#9648]
Looks similar to an issue that was posted here last year about unusual fields being required. 

I have an instance running in Ubuntu and never saw this. Setup a new on Windows Server 2019. I’ve setup everything but assets and had no issues. First asset I try and it requires a supplier or can’t create. 

“The selected supplier ID is invalid”

No supplier was added; none of the fields even adjacent were interacted with.

**Comment by alockbox** [comment]
Snipe-IT version v5.1.5 build 6029 (gb8c356443)
PHP version 8.0.6
Laravel version 6.20.22
OS Windows Server 2019

Installation was with this guide: https://blog.masteringmdm.com/snipe-it-installation-on-windows-server/

Since my asset models all had custom fields. I created barebones test asset model, test category that has none of that. Still fails with error “The selected supplier id is invalid”.

The only thing I can even think that I did which was non-standard was pointing the PHP.ini to gd since gd2 was removed with newer php. 

Only other thing I’ve noted is to not use IE included with server 2019 (IE v1809) as custom fields do not show up when an asset model that requires them is selected. This works properly in Chrome.

I enabled logging but see nothing pertinent in the logs, let me know if there’s something specific I should copy/paste here.

**Comment by alockbox** [comment]
Ok, if I create supplier and then select that, it allows me to finally add the asset. No other “unrequired” fields become required. I was worried it would be a chain of one after the other but it doesn’t seem to be the case. 

To reiterate, supplier field is exhibiting as Required although it is not.

**Comment by alockbox** [comment]
Weirder still, If I go back in to the asset, I can now remove the supplier and save with no errors. 

On another note, my search function in the dropdowns does not work for users. I’m only able to pick the name not search for a partial name match. This seems to only happen with users (14 users so far), I can search other fields likes assets (30+). Here I thought I’d have an easier time on Windows!

I can upload the laravel log if that helps but it seems to have person items in it. I'm seeing a few invalid parameter number errors like this in my log:
[2021-05-30 01:25:54] production.ERROR: SQLSTATE[HY093]: Invalid parameter number (SQL: select count(*) as aggregate from `users` where (`show_in_list` = 1 and (`first_name` LIKE %wa% or `last_name` LIKE %wa% or CONCAT(users.first_name," ",users.last_name) LIKE %wa%) or `username` LIKE %wa% or `employee_num` LIKE %wa%) and `users`.`deleted_at` is null) {"userId":1,"exception":"[object] (Illuminate\\Database\\QueryException(code: HY093): SQLSTATE[HY093]: Invalid parameter number (SQL: select count(*) as aggregate from `users` where (`show_in_list` = 1 and (`first_name` LIKE %wa% or `last_name` LIKE %wa% or CONCAT(users.first_name,\" \",users.last_name) LIKE %wa%) or `username` LIKE %wa% or `employee_num` LIKE %wa%) and `users`.`deleted_at` is null) at C:\\inetpub\\wwwroot\\snipeit\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Connection.php:669)
[stacktrace]



Another line 
Next Illuminate\Database\QueryException: SQLSTATE[HY093]: Invalid parameter number (SQL: select count(*) as aggregate from `users` where (`show_in_list` = 1 and (`first_name` LIKE %andy% or `last_name` LIKE %andy% or CONCAT(users.first_name," ",users.last_name) LIKE %andy%) or `username` LIKE %andy% or `employee_num` LIKE %andy%) and `users`.`deleted_at` is null) in C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php:669

**Comment by alockbox** [comment]
Well, I’m proud of myself a bit and peev’ed at myself a bit. It turns out you all clearly say don’t use php8 but I somehow missed that with the guide I used. 

I reused the guide for the php sections, doing everything again for 7.4.x. 

Everything works perfectly now. 

I could swear it’s actually faster. 

So the only bugs I experienced on a non-supported php was those two things! Nice job guys!!!

thanks for a wonderful app.
